src/containers/Resources/index.jsx

```diff
@@ -1,18 +1,119 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import SearchBar from "../../components/SearchBar";
 import NgoList from "../../components/NgoList";
 import AdvancedFilters from "../../components/AdvancedFilters";
 import PagesList from "../../components/PagesList";
+import database from "../../firebaseConfig";
 
 const Resources = () => {
+  const [allNgos, setAllNgos] = useState([]);
+  const [filteredNgos, setFilteredNgos] = useState([]);
+  const [filterOptions, setFilterOptions] = useState([]);
+  const [searchKey, setSearchKey] = useState("");
+
+  useEffect(() => {
+    // Fetch all ngos from the database
+    const fetchNgos = (setNgos) => {
+      database
+        .collection("ngosEn")
+        .get()
+        .then((querySnapshot) => {
+          querySnapshot.forEach((doc) => {
+            setNgos((prevState) => [
+              ...prevState,
+              { id: doc.id, ngoData: doc.data() },
```

Comment:
there's extra nesting here, when I was helping you with function i changed it into {...doc.data()}, this one works too but the functions are working fine right ?